request/request.js

```diff
@@ -115,19 +115,38 @@ module.exports = function($window, Promise) {
 				if (xhr.readyState === 4) {
 					try {
 						var success = (xhr.status >= 200 && xhr.status < 300) || xhr.status === 304 || (/^file:\/\//i).test(url)
-						var response = xhr.responseText
+						// When the response type isn't "" or "text",
+						// `xhr.responseText` is the wrong thing to use.
+						// Browsers do the right thing and throw here, and we
+						// should honor that and do the right thing by
+						// preferring `xhr.response` where possible/practical.
+						var response = xhr.response, message
+
+						if (response == null) {
+							try {
+								response = xhr.responseText
+								// Note: this snippet is intentionally *after*
```

Comment:
This fallback also needs verified in Edge. I don't have a VM readily available and set up for this, so it'd be appreciated if someone can confirm this works for that, too. (And if so, the comment needs updated to account for Edge's potential lack of support.)

Reply:
I tried your branch out:
![image](https://user-images.githubusercontent.com/848652/56688434-b774cc00-66a6-11e9-8eac-d4d32cfd7f52.png)

![image](https://user-images.githubusercontent.com/848652/56688425-b0e65480-66a6-11e9-9368-934c4448cee8.png)

Seems to work. 

I also tried hitting the case where response is null. Edge also throws and skips the logic when you try to assign response. 

![image](https://user-images.githubusercontent.com/848652/56688867-c1e39580-66a7-11e9-8aa4-addaea560fab.png)

Reply:
Did you verify it works after rebuilding the bundle? (You may need to create a local fork of my branch, publish it temporarily, and then point Flems to the raw link to avoid hitting Flems' character limit.)

Reply:
I cloned your fork, checked out this branch, ran `npm i` and `npm run build`

I then tried with mithril.js (with the comments in the screenshot you see) and mithril.min.js, both worked, for the simple `m.request` test I did. 

It fit within flems' character limit.

Reply:
Okay.

And what I meant with the "character limit" was that I wasn't sure if Flems would complain or bork if the hash got truncated in the URL or anything like that. If it works, it works.